pingcap/tidb-operator
# Recreate sts after resize all PVCs

### What problem does this PR solve?

Close #4547
### What is changed and how does it work?


### Code changes


### Tests



For every component:
1. Scale up the storage size from 200Gi to 300Gi.
2. `Resizing` condition is set to true.
3. Check that PVCs are resizing pod by pod.
4. Check that STS is recreated after all PVCs are resized.
5. `Resizing` condition is set to false.
6. Scale out replication, check the size of new PVC is expected.

For storage class not support expansion:
1. Scale up the storage size for TiKV from 200Gi to 300Gi.
2. `Resizing` condition is set to true.
4. Check STS is recreated with 200Gi size, but PVCs are not changed.
7. `Resizing` condition is set to false.
8. Scale out TiKV, check the size of  new PVC is 300Gi.



### Side effects


### Related changes


### Release Notes

Please refer to [Release Notes Language Style Guide](https://github.com/pingcap/tidb-operator/blob/master/docs/release-note-guide.md) before writing the release note.

```release-note
Recreate statefulset after resize all PVCs for a components
```
